code-studio/src/js/components/progress/course_progress.jsx

```diff
@@ -2,15 +2,25 @@
 
 var STAGE_TYPE = require('./types').STAGE_TYPE;
 var CourseProgressRow = require('./course_progress_row');
+var StageDetails = require('./stage_details');
 
 /**
  * Stage progress component used in level header and course overview.
  */
 var CourseProgress = React.createClass({
   propTypes: {
+    display: React.PropTypes.oneOf(['dots', 'list']).isRequired,
     stages: React.PropTypes.arrayOf(STAGE_TYPE)
   },
 
+  getRow: function (stage) {
+    if (this.props.display === 'dots') {
+      return <CourseProgressRow stage={stage} key={stage.name}/>;
+    } else {
+      return <StageDetails  stage={stage} key={stage.name}/>;
```

Comment:
I'm down with that, since we'd then [conform with airbnb](https://github.com/airbnb/javascript/tree/master/react#spacing).  I think @Bjvanminnen was reluctant to lock thisrule down last time we talked about this though.